exchanges/kucoin/kucoin_test.go

```diff
@@ -0,0 +1,2505 @@
+	}
+}
+func TestUpdateTicker(t *testing.T) {
+	t.Parallel()
+	newP, err := currency.NewPairFromString("MKR-USDT")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = ku.UpdateTicker(context.Background(), newP, asset.Spot)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestFetchTicker(t *testing.T) {
+	t.Parallel()
+	enabledPairs, err := ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err = ku.FetchTicker(context.Background(), enabledPairs[0], asset.Spot); err != nil {
+		t.Fatal(err)
+	}
+	if _, err = ku.FetchTicker(context.Background(), enabledPairs[0], asset.Margin); err != nil {
+		t.Error(err)
+	}
+	enabledPairs, err = ku.GetEnabledPairs(asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err = ku.FetchTicker(context.Background(), enabledPairs[len(enabledPairs)-1], asset.Futures); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestFetchOrderbook(t *testing.T) {
+	t.Parallel()
+	enabledPair, err := ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err := ku.FetchOrderbook(context.Background(), enabledPair[0], asset.Spot); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetHistoricCandles(t *testing.T) {
+	enabledPairs, err := ku.GetEnabledPairs(asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+	startTime := time.Now().Add(-time.Hour * 4)
+	endTime := time.Now().Add(-time.Hour * 3)
+	_, err = ku.GetHistoricCandles(context.Background(), enabledPairs[len(enabledPairs)-1], asset.Futures, kline.OneHour, startTime, endTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+	enabledPairs, err = ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = ku.GetHistoricCandles(context.Background(), enabledPairs[len(enabledPairs)-1], asset.Spot, kline.OneHour, startTime, time.Now())
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetHistoricCandlesExtended(t *testing.T) {
+	enabledPairs, err := ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Fatal(err)
+	}
+	startTime := time.Now().Add(-time.Hour * 4)
+	endTime := time.Now().Add(-time.Hour * 1)
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), enabledPairs[0], asset.Spot, kline.OneHour, startTime, endTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), enabledPairs[0], asset.Spot, kline.FiveMin, startTime, endTime)
+	if err != nil {
+		t.Error(err)
+	}
+	enabledPairs, err = ku.GetEnabledPairs(asset.Margin)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), enabledPairs[0], asset.Margin, kline.OneHour, startTime, endTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tradablePairs, err := ku.FetchTradablePairs(context.Background(), asset.Futures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), tradablePairs[0], asset.Futures, kline.FiveMin, startTime, endTime)
+	if err != nil && !errors.Is(err, kline.ErrNoTimeSeriesDataToConvert) {
+		t.Error(err)
+	}
```

Comment:
I used an element from the tradable pairs because I thought it had to work for all instruments, but when it kept returning the error message you saw, I left an error check to show that this endpoint request may not work for all requests. But I won't use that if you think it's a bad approach.